fe/fe-core/src/main/java/com/starrocks/connector/unified/UnifiedMetadata.java

```diff
@@ -0,0 +1,213 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.connector.unified;
+
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.MaterializedIndexMeta;
+import com.starrocks.catalog.PartitionKey;
+import com.starrocks.catalog.Table;
+import com.starrocks.common.AlreadyExistsException;
+import com.starrocks.common.DdlException;
+import com.starrocks.common.MetaNotFoundException;
+import com.starrocks.common.Pair;
+import com.starrocks.connector.ConnectorMetadata;
+import com.starrocks.connector.PartitionInfo;
+import com.starrocks.connector.RemoteFileInfo;
+import com.starrocks.connector.hive.HiveMetadata;
+import com.starrocks.sql.ast.CreateTableStmt;
+import com.starrocks.sql.ast.DropTableStmt;
+import com.starrocks.sql.optimizer.OptimizerContext;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.statistics.Statistics;
+import com.starrocks.thrift.TSinkCommitInfo;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.starrocks.catalog.Table.TableType.DELTALAKE;
+import static com.starrocks.catalog.Table.TableType.HIVE;
+import static com.starrocks.catalog.Table.TableType.HUDI;
+import static com.starrocks.catalog.Table.TableType.ICEBERG;
+import static java.util.Objects.requireNonNull;
+
+public class UnifiedMetadata implements ConnectorMetadata {
+    public static final String ICEBERG_TABLE_TYPE_NAME = "table_type";
+    public static final String ICEBERG_TABLE_TYPE_VALUE = "iceberg";
+    public static final String SPARK_TABLE_PROVIDER_KEY = "spark.sql.sources.provider";
+    public static final String DELTA_LAKE_PROVIDER = "delta";
+
+    static boolean isIcebergTable(Map<String, String> properties) {
+        return ICEBERG_TABLE_TYPE_VALUE.equalsIgnoreCase(properties.get(ICEBERG_TABLE_TYPE_NAME));
+    }
+
+    static boolean isDeltaLakeTable(Map<String, String> properties) {
+        return DELTA_LAKE_PROVIDER.equalsIgnoreCase(properties.get(SPARK_TABLE_PROVIDER_KEY));
+    }
+
+    private final Map<Table.TableType, ConnectorMetadata> metadataMap;
+    private final HiveMetadata hiveMetadata; // used to determine table type
+
+    public UnifiedMetadata(Map<Table.TableType, ConnectorMetadata> metadataMap) {
+        requireNonNull(metadataMap, "metadataMap is null");
+        checkArgument(metadataMap.containsKey(HIVE), "metadataMap does not have hive metadata");
+        this.metadataMap = metadataMap;
+        this.hiveMetadata = (HiveMetadata) metadataMap.get(HIVE);
+    }
+
+    private Table.TableType getTableType(String dbName, String tblName) {
+        Table table = hiveMetadata.getTable(dbName, tblName);
```

Comment:
if tblname is a iceberg/delta lake table, hiveMetadata.getTable could get iceberg/delta lake table?

Reply:
Yes. Use table properties to determine table type.